DisCatSharp.ApplicationCommands/ApplicationCommandsExtension.cs

```diff
@@ -638,6 +637,24 @@ private void RegisterCommands(IEnumerable<ApplicationCommandsModuleConfiguration
 										});
 									}
 								}
+								else
+								{
+									var (success, commandId, permissions) = PermissionWorker.ResolveGlobalPermissions(types, command.Id, com.Name, source.Value, source.Key);
+
+									if (success)
+									{
+										var perms = permissions.Select(x => x.Key).Distinct();
+										foreach(var guild in perms)
+										{
+											guildOverwrites.Add(new KeyValuePair<ulong, DiscordGuildApplicationCommandPermission>(guild, new DiscordGuildApplicationCommandPermission()
+											{
+												Id = commandId.Value,
+												Permissions = permissions.Where(x => x.Key == guild).Select(x => x.Value).ToList(),
+												GuildId = guild
+											}));
+										}
+									}
```

Comment:
I notice this block of code is nearly identical to other spots. This feels like it should be a function of sorts

Reply:
Agreed, we could cut it down to something like: (this is a mix of C++ and C#)

```
void OverwriteNewPerms([&bot], int* commandId, ptr::vector* permissions) {
    var (success, commandId, permissions) = PermissionsWorker.ResolveGlobalPermissions(types, command.id, com.Name, source.Value, source.Key);

    if (success) 
    {
          var perms = permissions.Select(x => x.Key).Distinct();
          foreach (var guild in perms)
          {
               // Here i cba writing the keyPair stuff
          }
     } else {
         Console.Write(/*the error code*/);
    }
};
```

Reply:
hmm